src/metabase/db/data_migrations.clj

```diff
@@ -369,6 +369,19 @@
                                  [:like
                                   :dashcard.visualization_settings "%\"click_link_template\":%"]]})))
 
+(defmigration ^{:author "jeff303" :added "0.43.0"} remove-bigquery-driver
+  (doseq [{:keys [:details] :as bigquery-db} (db/select Database :engine "bigquery")]
+    (when-not (empty? (filter some? ((juxt :auth-code :client-id :client-secret) details)))
+      (log/errorf (str "Database ID %d used the :bigquery driver with OAuth style authentication scheme, which cannot"
+                       " be automatically migrated to :bigquery-cloud-sdk; leaving authentication options nil in the"
+                       " db details, which must be corrected by an admin (by setting a service-account-json), so that"
+                       " sync and query can work again")
+                  (u/the-id bigquery-db)))
+    (let [migrated-details (-> (assoc details :engine :bigquery-cloud-sdk) ; switch to new driver
+                               ;; remove all OAuth related keys
+                               (dissoc :auth-code :client-id :client-secret :access-token :refresh-token))]
+      (db/update! Database (u/the-id bigquery-db) :details migrated-details))))
```

Comment:
I really really would prefer this be done in a Liquibase migration. Are you 100% sure we can't just migrate the driver in Liqubase e.g.

```sql
UPDATE metabase_database
SET engine = 'bigquery-cloud-sdk'
WHERE engine = 'bigquery'
```

And then detect and correct this situation (legacy OAuth details) in the `normalize-db-details` method?

Reply:
Actually, wouldn't it be sufficient just to ignore those keys entirely? Do you even need to remove them at all?

Reply:
Yeah, this is feasible.  I originally got stuck into thinking I could register `normalize-db-details` for `:bigquery`, but that obviously wouldn't work if the driver (which needs initialization) is being removed.